### PR TITLE
contracts-bedrock: portal opaque data test

### PR DESCRIPTION
**Pull request by tynes**
**Description**

- **contracts-bedrock: add deposit event coverage**
- **contracts-bedrock: revert diff to portal**

contracts-bedrock: add deposit event coverage

Adds deposit event differential coverage for the two
different places where the deposit tx event exists.
This ensures that a valid event ends up getting emitted.

A commit 372ec6bd0ae4fd1d23db74bffa5eb7335ed4dbeb
is reverted that modularizes the code for serializing
the event so that there are not footguns when computing
the "opaque data". We can bring this back easily if
we would like.

### CODE REVIEW

File: packages/contracts-bedrock/test/L1/OptimismPortal.t.sol [hunk 1]
```diff
@@ -470,6 +471,33 @@ contract OptimismPortal_Test is CommonTest {
...
     }
 
+    /// @notice Ensures that the deposit event is correct for the `setGasPayingToken`
+    ///         code path that manually emits a deposit transaction outside of the
+    ///         `depositTransaction` function. This is a simple differential test.
+    function test_setGasPayingToken_correctEvent_succeeds() external {
+        vm.recordLogs();
+
+        vm.prank(address(systemConfig));
+        optimismPortal.setGasPayingToken({ _token: address(0x9999), _decimals: 19, _name: "", _symbol: "" });
```
Comment (geoknee) [inline packages/contracts-bedrock/test/L1/OptimismPortal.t.sol]: Would it be better to use nonzero values for name and symbol, just to avoid any edge cases where something isn't being set somewhere, and also to exercise maximum gas expenditure (since we now know that can be an issue)?

**Review by geoknee — APPROVED** [review]
LGTM, left one suggestion.

File: packages/contracts-bedrock/test/L1/OptimismPortal.t.sol (hunk 1 above)
Comment (maurelian) [reply packages/contracts-bedrock/test/L1/OptimismPortal.t.sol]: Moreover, this test could be made stronger by making it a fuzz test with symbolic values passed into `setGasPayingToken()`.

File: packages/contracts-bedrock/test/L1/OptimismPortal.t.sol [hunk 2]
```diff
@@ -470,6 +471,33 @@ contract OptimismPortal_Test is CommonTest {
...
+            _data: abi.encodeCall(L1Block.setGasPayingToken, (address(0x9999), 19, bytes32(0), bytes32(0)))
+        });
+
+        VmSafe.Log[] memory logs = vm.getRecordedLogs();
+        assertEq(logs.length, 2);
+
+        VmSafe.Log memory systemPath = logs[0];
+        VmSafe.Log memory userPath = logs[1];
+
+        assertEq(systemPath.data, userPath.data);
```
Comment (maurelian) [inline packages/contracts-bedrock/test/L1/OptimismPortal.t.sol:516]: I think it'd be a good idea to also compare the `topics` fields

File: packages/contracts-bedrock/test/L1/OptimismPortal.t.sol (hunk 1 above)
Comment (tynes) [reply packages/contracts-bedrock/test/L1/OptimismPortal.t.sol]: I have turned this into a fuzz test, you are right @maurelian

File: packages/contracts-bedrock/test/L1/OptimismPortal.t.sol (hunk 2 above)
Comment (tynes) [reply packages/contracts-bedrock/test/L1/OptimismPortal.t.sol:516]: I have updated to also compare topics fields

File: packages/contracts-bedrock/src/L1/OptimismPortal.sol [hunk 3]
```diff
@@ -128,8 +128,8 @@ contract OptimismPortal is Initializable, ResourceMetering, ISemver {
     }
 
     /// @notice Semantic version.
-    /// @custom:semver 2.8.0
-    string public constant version = "2.8.0";
+    /// @custom:semver 2.7.0
+    string public constant version = "2.7.0";
```
Comment (maurelian) [inline packages/contracts-bedrock/src/L1/OptimismPortal.sol]: Why is this being reduced?

Comment (tynes) [reply packages/contracts-bedrock/src/L1/OptimismPortal.sol]: This is an artifact of the rebase, apologies

**Review by maurelian — APPROVED** [review]
